### DISCUSSION TITLE
STM32 - ILI9486 4" SPI display

### DISCUSSION THREAD

**Discussion by maciekish** [Bodmer/TFT_eSPI#2560] [Q&A - General]
Hi, I have an [LCDWIKI 4" SPI display based on ILI9486](http://www.lcdwiki.com/4.0inch_SPI_Module_ILI9486). Trying to run it on an STM32F411CEU6 with Arduino IDE. All i get is a white screen. I have tried the config below. Triple checked the wiring. Also tried 18Mhz SPI frequency without luck.

```
// See SetupX_Template.h for all options available
#define USER_SETUP_ID 303

#define STM32
//#define ILI9486_DRIVER // Tried a lot of combinations of these, none seem to work.
//#define RPI_ILI9486_DRIVER
#define RPI_DISPLAY_TYPE
#define ILI9486_DRIVER

#define TFT_WIDTH  320
#define TFT_HEIGHT 480

#define TFT_CS   PA4
#define TFT_DC   PA3
#define TFT_RST  PA2
#define TFT_MOSI PA7
#define TFT_MISO PA6
#define TFT_SCLK PA5

#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts

// STM32 support for smooth fonts via program memory (FLASH) arrays
#define SMOOTH_FONT

// Assuming the processor clock is 72MHz:
#define SPI_FREQUENCY  36000000   // 36MHz SPI clock
//#define SPI_FREQUENCY  18000000   // 18MHz SPI clock

#define SPI_READ_FREQUENCY  12000000 // Reads need a slower SPI clock

#define SPI_TOUCH_FREQUENCY  2500000 // Must be very slow

```

**Comment by Bodmer** [comment]
Comment out or delete the following line as it is not an RPi display.
#define RPI_DISPLAY_TYPE

**Reply by maciekish** [reply]
Sorry should have been clearer. I tried using only #define ILI9486_DRIVER as well, also white screen. Cant get it to work on a Mega using the “official” lib either. Could the display be defective? Its new but Aliexpress QC isnt the best.

**Reply by Bodmer** [reply]
It wont work with a Mega unless 5V to 3.3V logic converters are used. The display requires 3.3V logic signals and will not initialise correctly with 5V signals due to the signal overload.

**Reply by Bodmer** [reply]
Try using the signal wiring and setup as defined here:
https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setups/Setup32_ILI9341_STM32F103.h

Just change the driver to #define ILI9486_DRIVER

**Comment by Bodmer** [comment]
I have connected up an STM32F411 "Black Pill" board and have found that the setup 32 does not work. I suspect that this is due to updates in the STM32 board package as the Black Pill boards no longer feature in the selection menu. Instead generic STM32F4xx board has to be selected.  This change to the board package also means that the SPI pins must be defined and the STM format pin references used,

Here is a working setup for my board with an ILI9431 display. It works fine. I do not have an ILI9486 SPI board to test.

The setup worked with the latest 2.0.x IDE. Some sketches require the serial port to be enabled and a monitor window opened before they run, so watch out for that. Here is the setup file I used:
```
        ///////////////////////////////////////////////////////////////
        // Setup for STM32F411 (e.g. Black Pill) and ILI9341 display //
        ///////////////////////////////////////////////////////////////
#define USER_SETUP_ID 32

#define STM32

#define ILI9341_DRIVER
//#define ILI9486_DRIVER

#define TFT_CS   PA4
#define TFT_DC   PA3
#define TFT_RST  PA2

#define TFT_MOSI PA7
#define TFT_MISO PA6
#define TFT_SCLK PA5

#define LOAD_GLCD
#define LOAD_FONT2
#define LOAD_FONT4
#define LOAD_FONT6
#define LOAD_FONT7
#define LOAD_FONT8
#define LOAD_GFXFF

#define SMOOTH_FONT

#define SPI_FREQUENCY  36000000   // 36MHz SPI clock
//#define SPI_FREQUENCY  18000000   // 18MHz SPI clock

#define SPI_READ_FREQUENCY  12000000

#define SPI_TOUCH_FREQUENCY  2500000

```

IDE setting used:
![image](https://user-images.githubusercontent.com/15824805/230999856-cc26ed44-a89c-46ca-a1ac-a2824fc1d7bc.png)

![image](https://user-images.githubusercontent.com/15824805/231000098-3d1c6161-869b-4aa0-8aaf-4800d76a1b65.png)
